pkg/controllers/termination/suite_test.go

```diff
@@ -204,9 +204,17 @@ var _ = Describe("Termination", func() {
 			podNoEvict := test.Pod(test.PodOptions{
 				NodeName:   node.Name,
 				ObjectMeta: metav1.ObjectMeta{Labels: labelSelector},
+				Phase:      v1.PodRunning,
 			})
 
-			ExpectCreated(ctx, env.Client, node, podNoEvict, pdb)
+			ExpectCreated(ctx, env.Client, node, pdb)
+			ExpectCreatedWithStatus(ctx, env.Client, podNoEvict)
+
+			// kube-controller-manager normally does this, but we don't have one. If this isn't modified
+			// the eviction controller assumes that the PDB hasn't been processed by the disruption controller
+			// yet and adds a 10 second retry to our evict() call
+			pdb.Status.ObservedGeneration = 1
```

Comment:
Can we include this in the `test.PodDisruptionBudget` library? We will then need to use ExpectCreatedWithStatus on it.

Reply:
Yup, done, the update call required the `ResourceVersion` to match, I thought that was with everything so I'm not sure why it only showed up for the PDB.